File: write-api-server-kotlin/src/main/kotlin/com/github/j5ik2o/adceet/api/write/adaptor/http/ExceptionHandlers.kt
package com.github.j5ik2o.adceet.api.write.adaptor.http

import akka.http.javadsl.model.StatusCodes
import akka.http.javadsl.server.Directives.complete
import akka.http.javadsl.server.ExceptionHandler
import com.github.j5ik2o.adceet.api.write.use.case.ThreadException

object ExceptionHandlers {

  val defaultHandler: ExceptionHandler
    get() {
      val builder = ExceptionHandler.newBuilder()
      builder.match(ThreadException::class.java) { ex ->
        complete(StatusCodes.BAD_REQUEST, ex.message)
      }
      return builder.build()
    }
}

File: write-api-server-scala/src/main/scala/com/github/j5ik2o/adceet/api/write/http/ExceptionHandlers.scala
package com.github.j5ik2o.adceet.api.write.http

import akka.http.scaladsl.model.StatusCodes
import akka.http.scaladsl.server.ExceptionHandler
import com.github.j5ik2o.adceet.api.write.use.`case`.ThreadException
import akka.http.scaladsl.server.Directives._

object ExceptionHandlers {

  val defaultHandler: ExceptionHandler = ExceptionHandler { case ex: ThreadException =>
    complete(StatusCodes.BadRequest, ex.message)
  }

}
